omod/src/main/java/org/openmrs/module/providermanagement/rest/search/ProviderSearchHandler.java

```diff
@@ -0,0 +1,70 @@
+package org.openmrs.module.providermanagement.rest.search;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.providermanagement.Provider;
+import org.openmrs.module.providermanagement.ProviderRole;
+import org.openmrs.module.providermanagement.api.ProviderManagementService;
+import org.openmrs.module.webservices.rest.web.RequestContext;
+import org.openmrs.module.webservices.rest.web.RestConstants;
+import org.openmrs.module.webservices.rest.web.resource.api.PageableResult;
+import org.openmrs.module.webservices.rest.web.resource.api.SearchConfig;
+import org.openmrs.module.webservices.rest.web.resource.api.SearchHandler;
+import org.openmrs.module.webservices.rest.web.resource.api.SearchQuery;
+import org.openmrs.module.webservices.rest.web.resource.impl.EmptySearchResult;
+import org.openmrs.module.webservices.rest.web.resource.impl.NeedsPaging;
+import org.openmrs.module.webservices.rest.web.response.ResponseException;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+@Component
+public class ProviderSearchHandler implements SearchHandler {
+
+    protected Log log = LogFactory.getLog(getClass());
+
+    protected final String PROVIDER_ROLES_PARAM = "providerRoles";
+
+    private final SearchConfig searchConfig = new SearchConfig("providerByRole", RestConstants.VERSION_1 + "/provider",
+            Arrays.asList("1.9.* - 9.*"),
+            new SearchQuery.Builder(
+                    "Allows you to find providers by provider role uuid").withRequiredParameters(PROVIDER_ROLES_PARAM).build());
+
+    /**
+     * @see org.openmrs.module.webservices.rest.web.resource.api.SearchHandler#getSearchConfig()
+     */
+    @Override
+    public SearchConfig getSearchConfig() {
+        return searchConfig;
+    }
+
+    /**
+     * @see org.openmrs.module.webservices.rest.web.resource.api.SearchHandler#search(org.openmrs.module.webservices.rest.web.RequestContext)
+     */
+    @Override
+    public PageableResult search(RequestContext context) throws ResponseException {
+  
+        String providerRoleUuids = context.getParameter(PROVIDER_ROLES_PARAM);
+        List<ProviderRole> providerRoles = new ArrayList<ProviderRole>();
+
+        for (String providerRoleUuid : providerRoleUuids.split(",")) {
+            ProviderRole providerRole = Context.getService(ProviderManagementService.class).getProviderRoleByUuid(providerRoleUuid);
+            if (providerRole != null) {
+                providerRoles.add(providerRole);
+            }
+            else {
+                log.warn("Unable to find provider role with uuid: " + providerRoleUuid);
```

Comment:
I don't think this should be a warning.  If we are asking for something specific, and that specific thing is invalid, I think this should fail with an error.  Otherwise, you get different data than was asked for.

Reply:
Fair, updated